game-core/src/main/java/games/strategy/triplea/ai/pro/AbstractProAi.java

```diff
@@ -86,24 +90,35 @@ public AbstractProAi(
     storedPurchaseTerritories = null;
     storedPoliticalActions = null;
     storedStrafingTerritories = new ArrayList<>();
+
+    // in case there are multiple Pro AIs running at the same time, unregister the one
+    // created before creating one so that there is only one menu item for all of the ProAI
+    DebugMenu.unregisterMenuCallback("Hard AI");
+    DebugMenu.registerMenuCallback("Hard AI", this::initialize);
   }
 
   @Override
   public void stopGame() {
     super.stopGame(); // absolutely MUST call super.stopGame() first
     calc.stop();
+
+    DebugMenu.unregisterMenuCallback("Hard AI");
   }
 
   public ProOddsCalculator getCalc() {
     return calc;
   }
 
-  public static void initialize(final TripleAFrame frame) {
+  private Collection<JMenuItem> initialize(final TripleAFrame frame) {
```

Comment:
The frame dependency appears to only be here due to the `initialize` call. It does not look that is related to the menu item. Could we simply move the initialize call up a level and avoid the `frame` parameter?

Reply:
It is related to the action of the menu item.  It can't be moved up since each menu item might be doing different things and they need access to the TripleAFrame so they can act on it.

Reply:
To walk you through why it is needed as part of the action, when you click on the link, it will call `AbstractProAi::showSettingsWindow`.  And that method calls `ProLogUi.showSettingsWindow` which calls `settingsWindow.setVisible(true)`.  And `settingsWindow' was created in the `initialize` method.  So, without the `TripleAFrame`, the action wouldn't be able to function.  Unless you passed in `ProLogWindow` instead of `TripleAFrame` and now the debug factory is dependent on `ProLogWindow` instead of `TripleFrame` and you are limited to what `ProLogWindow` can do.